### ISSUE TITLE
Two postal instance on same domain ?

### ISSUE THREAD

**Issue by guldil** [postalhq/postal#1077]
Hy,

It's not a bug just a question. Is it possible to create two postal instance (with own MySQL and RabitMQ) to handle same domain ? It's for redundacy, if i need to reboot postal ?
As a Active / Passive configuration.
I check quickly the cluster RabbitMQ + Cluster MySQL but it took too much ressources...

First instance : 
 mx_records:
    - mail.yourdomain.com
  smtp_server_hostname: mail.yourdomain.com
  spf_include: spf.mail.yourdomain.com
  return_path: rp.mail.yourdomain.com
  route_domain: routes.mail.yourdomain.com
  track_domain: track.mail.yourdomain.com
  dkim_identifier: postal
  domain_verify_prefix: postal-verification
  custom_return_path_prefix: psrp

Second instance : 

 mx_records:
    - mail-backup.yourdomain.com
  smtp_server_hostname: mail-backup.yourdomain.com
  spf_include: spf.mail-backup.yourdomain.com
  return_path: rp.mail-backup.yourdomain.com
  route_domain: routes.mail-backup.yourdomain.com
  track_domain: track.mai-backupl.yourdomain.com
  dkim_identifier: postal
  domain_verify_prefix: postal-verification
  custom_return_path_prefix: psrp

I know that if i send a mail through master i could loose webhook if it reboot just after.

Thanks for your time !

Guldil

**Comment by willpower232** [comment]
You will be interested in https://github.com/postalhq/postal/issues/694#issuecomment-435818607

TLDR, If you're concerned about redundancy, it would be easier to have Postal with the same config on multiple servers pointing at the same databases and rabbitmq

**Comment by guldil** [comment]
Hello willpower232,

I already have a Mysql Cluster (Percona) but i have no RabbitMQ knowledges so it could be difficult to install it in cluster mode just for Postal. Each Postal instance can't use her own RabbitMQ server with a centralized database ?

**Comment by willpower232** [comment]
I'm sure you can give it a go but I'm not sure how that would affect messages actually being sent if you lost one of the queues.

**Comment by guldil** [comment]
We tried to install multiple postal instance with a same database, it must "initialize" MySQL DB. After second postal install first one is broken...

**Comment by vishnus** [comment]
> TLDR, If you're concerned about redundancy, it would be easier to have Postal with the same config on multiple servers pointing at the same databases and rabbitmq

Hi @willpower232 could this speed up the queuing process as well? We faced this issue when we try to send massive amounts of mails with attachments. The queuing was done by single process and hence it slowed down the whole process. Trying to figure out if running the Postal app in multiple servers behind loadbalancer pointing to same DB and RabbitMq could solve this issue...

**Comment by willpower232** [comment]
@guldil for the secondary servers you would need to manually follow some slightly different steps to avoid resetting the database, I think its a matter of installing ruby, copying the code and config in, installing the gems and dependencies, and generating the assets for the UI.

@vishnus I presume you are sending each message to Postal individually from your application so a load balanced setup might help you out here. Your other option would be to slow down the messages being sent from your code in whatever loop is happening.

**Comment by vishnus** [comment]
> @guldil for the secondary servers you would need to manually follow some slightly different steps to avoid resetting the database, I think its a matter of installing ruby, copying the code and config in, installing the gems and dependencies, and generating the assets for the UI.

If we take an image of the existing postal instance, spin off another instance, and before starting Postal process, we switch off DB and RabbitMQ and change the Postal config to DB and RabbitMQ in server A, and remove cron, requeuer, worker and web processes from server B and only run SMTP, would that work in my case? And in case of @guldil he would have to run all processes?

**Comment by willpower232** [comment]
Yes that sounds like it would work nicely, as long as you're running all the processes somewhere and they all have access to RabbitMQ and MySQL

**Comment by vishnus** [comment]
@willpower232 so that worked beautifully. We ran 20 tiny c5.large instances with only SMTP process running and put them all behind a network loadbalancer and connected all to same big Postal instance's DB and RabbitMq and we were able to send over half million emails with total attachments size of 40+GB in just 1 hour :) Thanks

@guldil this set up will work. We needed only SMTP to run, hence we disabled other apps from the ProcFile. In your case, you can run all processes in the new server, connect to the main DB and RabbitMq, and put all instances behind a NLB.
